src/storage/src/source/reclock.rs

```diff
@@ -340,50 +339,56 @@ where
     }
 
     pub fn compact(&mut self, new_since: Antichain<IntoTime>) {
-        let mut inner = &mut *self.inner.borrow_mut();
         // Ignore compaction requests while we initialize
-        if !PartialOrder::less_than(&inner.as_of, &inner.upper) {
+        if !self.initialized() {
             return;
         }
-        if !PartialOrder::less_equal(&inner.since, &new_since) {
+        let inner = &mut *self.inner.borrow_mut();
+        if !PartialOrder::less_equal(&self.since, &new_since) {
             panic!(
                 "ReclockFollower: `new_since` ({:?}) is not beyond \
                 `self.since` ({:?}).",
-                new_since, inner.since,
+                new_since, self.since,
             );
         }
+        inner.since.update_iter(
```

Comment:
We need to be careful that we _never_ add any fallible code after we override this since, otherwise we would get inconsistent state; probably worth a comment

Reply:
What do you mean by fallible code? This method doesn't return a `Result`

Reply:
thats what I mean! if we ever changed it to be fallible